ngraph/frontend/onnx_import/src/op/upsample.cpp

```diff
@@ -70,6 +75,73 @@ namespace ngraph
             }
 
             namespace set_1
+            {
+                OutputVector upsample(const onnx_import::Node& node)
+                {
+                    const auto height_scale = node.get_attribute_value<float>("height_scale");
+                    const auto width_scale = node.get_attribute_value<float>("width_scale");
+                    const auto mode = node.get_attribute_value<std::string>("mode", "nearest");
+                    check_mode_support(node, mode, 1);
+
+                    const auto interpolate_mode = convert_mode(mode);
+                    const auto zero_pad = std::vector<size_t>(1, 0);
+                    const auto transform_mode = (interpolate_mode == InterpolateMode::linear_onnx
+                                                     ? Transform_mode::asymmetric
+                                                     : Transform_mode::half_pixel);
+                    const auto attrs =
+                        ngraph::op::v4::Interpolate::InterpolateAttrs(interpolate_mode,
+                                                                      ShapeCalcMode::scales,
+                                                                      zero_pad,
+                                                                      zero_pad,
+                                                                      transform_mode);
+
+                    const auto data = node.get_ng_inputs().at(0);
+
+                    if (data.get_partial_shape().is_static())
+                    {
+                        auto output_shape = data.get_shape();
+                        const auto rank = output_shape.size();
+                        std::vector<float> scales(rank, 1.f);
+                        scales[rank - 1] = width_scale;
+                        scales[rank - 2] = height_scale;
+                        output_shape[rank - 1] = std::floor(output_shape[rank - 1] * width_scale);
+                        output_shape[rank - 2] = std::floor(output_shape[rank - 2] * height_scale);
+
+                        const auto output_shape_const = default_opset::Constant::create(
+                            element::u64, Shape({output_shape.size()}), output_shape);
+                        const auto scales_const = default_opset::Constant::create(
+                            ngraph::element::f32, Shape({scales.size()}), scales);
+
+                        return {std::make_shared<default_opset::Interpolate>(
+                            data, output_shape_const, scales_const, attrs)};
+                    }
```

Comment:
Will this code work from the reshapebility perspective?
We should avoid to hard code shapes AFAIK.